### ISSUE TITLE
PyTorch Version

### ISSUE THREAD

**Issue by XiaohangZhan** [junyanz/iGAN#27]
Thank you for your excellent work. I believe this code will be more readable and extensible if in PyTorch. So will the PyTorch version be released? A link of third-party implementation is also fine, though I cannot find one for now.
Specifically, I'm interested in latent vector prediction and projection section.

**Comment by junyanz** [comment]
We are working on a PyTorch version. Hopefully, It will be released in summer.

**Comment by zzzzzzian** [comment]
Thank you in advance! It would be really great if PyTorch version is released! I am trying to apply iGAN to a project and the reading of code from Theano is painful. I find it hard to understand the constraint & manifold gradient descent part from the code. Is it possible to enlighten me a little about that part? I am also using PyTorch

**Comment by junyanz** [comment]
Yeah, I also forgot how to write Theano code. 
This [function](https://github.com/junyanz/iGAN/blob/master/constrained_opt_theano.py#L71) implements the loss function.

**Comment by Lifedecoder** [comment]
Two years and a half passed and the pytorch version of iGAN still not released?😂

**Comment by junyanz** [comment]
Sorry that it keeps getting delayed. If you want to use it for face editing, you can find it in photoshop "smart portrait". 
We have some new work with open-source PyTorch code that improves iGAN: [pix2latent](https://github.com/minyoungg/pix2latent) (compatible with BigGAN/StyleGAN.) and [ganseeing](https://github.com/davidbau/ganseeing) (compatible with StyleGAN).

**Comment by Lifedecoder** [comment]
Got it, thanks!
